### ISSUE TITLE
New Feature Proposal

### ISSUE THREAD

**Issue by Pierrecoulon1** [sukeesh/Jarvis#821]
Hi everybody,
I am Pierre and I will invest my time in the next months in this project to help you guys develop it.
I looked at what this Jarvis could do right now and I think this was a good idea to add a feature where you could ask for definition of words.
Thanks to the wikipedia library it’s quite easy to do, what do you think about it ?

**Comment by pnhofmann** [comment]
Hi!

> I am Pierre and I will invest my time in the next months in this project to help you guys develop it.

Always nice to hear ;).

> to add a feature where you could ask for definition of words

Sure! Just note: There is a Wikipedia-Plugin:
https://github.com/sukeesh/Jarvis/blob/master/jarviscli/plugins/wiki.py

And a dictionary plugin - but a quite basic one which works offline and displays stuff from ntlk:
https://github.com/sukeesh/Jarvis/blob/master/jarviscli/plugins/dictionary.py

You could enhance both ;). For example make dictionary also print wikipedia-definition (if user is online?). And/Or add features to wikipedia-plugin.

**Comment by Pierrecoulon1** [comment]
The wiki part is quite fully implemented. I made a funny part where you can ask him what to do when you're bored. I am using the Bored API. While implementing the plugin the command isn't recognize, what do I need to do?

**Comment by Pierrecoulon1** [comment]
I made many new features during the night. I implemented few others fun APIs which try to detect where you're from or your age from your name. Furthermore, I did a last one which permit someone to ask information about a country, Jarvis responds with his capital, his region, his number of inhabitants, his size and his currency

**Comment by pnhofmann** [comment]
Nice ;).

> While implementing the plugin the command isn't recognize, what do I need to do to make it work ?

Can you give an example code? Actually everything marked with ``@plugin`` in ``jarviscli/plugins`` dir should be loaded...

**Comment by Pierrecoulon1** [comment]
It's ok, there was a problem linked to the fact that the file and function were beginning with a Capital letter.
What would you want now? I was thinking of a Spotify API, but it needs Oauth2.
What would you prefer to see in the next Jarvis features ?

**Comment by pnhofmann** [comment]
> I was thinking of a Spotify API, but it needs Oauth2.

Wouldn't be a problem, if the user needs to login first.

Big terms of service ;) https://developer.spotify.com/terms/ but sure, something with Spotify would be nice.

**Comment by CyberkidAdithya** [comment]
Hello Brothers, I am a CS student from India. As I was looking for some help in my Python Project I came up with this Git Repo, and luckily you guys seem to be working on the same project in which I have been working for the past few months. So if you don't mind, will you guys allow me to contribute to this project? I have never done any Open Source Contributions before and I am really excited about doing one. I love Open Source Softwares by the way. And I am a big fan of Python, Firefox and VLC. And to talk about my works in this domain, I had made some really useful functions for automation and stuff, and I am even working on Computer Vision and GUI development with Tkinter.
